File: src/main/java/com/EduTrack/EDUTRACK_Portfolio_MS/model/SchoolProfile.java
package com.EduTrack.EDUTRACK_Portfolio_MS.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SchoolProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String logo;
    private String phoneNumber;
    private String email;
    private String address;
    private String description;
}

File: frontend/src/app/core/models/school-profile.ts
export interface SchoolProfile {
    id?: number;
    logo: string;
    phoneNbr: string;
    address: string;
    name: string;
    email: string;
    description: string;
    hashCode?: boolean;
    canEqual?: boolean;
    toString?: boolean;
}

export interface SchoolProfileBuilder {
    id: number;
    name: string;
    logo: string;
    phoneNbr: string;
    address: string;
    description: string;
    email: string;
}